modules/gateway/peers.go

```diff
@@ -184,62 +177,79 @@ func (g *Gateway) managedAcceptConnNewPeer(conn net.Conn, remoteVersion string)
 		return err
 	}
 
+	// Attempt to add the peer to the node list. If the add is successful and
+	// the address is a local address, mark the peer as a local peer.
+	local := false
+	err = g.managedAddUntrustedNode(remoteAddr)
+	if err != nil && remoteAddr.IsLocal() {
+		local = true
```

Comment:
I don't know of any way of manipulating your remote address over TCP. Is this really possible?